### PR TITLE
Add DNS health checks support for GitHub Pages

### LINKED ISSUE #2745: Add support for DNS health checks on Pages.
Hi :wave:

Currently, the `RepositoriesService` doesn't support [getting DNS health checks for Github pages](https://docs.github.com/en/rest/pages#get-a-dns-health-check-for-github-pages).

@gmlewis, I would like to implement this feature, if you agree.

### PR DESCRIPTION
**Pull request by oslowalk**
Fixes #2745

Allows requesting a DNS health check for the CNAME configured for a repository's GitHub Pages.

### CODE REVIEW

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:50]:
```suggestion
// PagesDomain represents a domain associated with a GitHub Pages site.
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:61]:
```suggestion
	HasCNAMERecord         *bool   `json:"has_cname_record,omitempty"`
```

File: github/repos_pages.go
Comment (gmlewis) [inline github/repos_pages.go:82]:
```suggestion
// PagesHealthCheckResponse represents the response given for the health check of a GitHub Pages site.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @oslowalk !
Just a few minor tweaks, please.

**Review by gmlewis — APPROVED** [review]
Thank you, @oslowalk !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by valbeat — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @valbeat !
Merging.

**Pull request merged**
